io.c

```diff
@@ -10056,45 +10059,32 @@ advice_arg_check(VALUE advice)
 
 /*
  *  call-seq:
- *     ios.advise(advice, offset=0, len=0) -> nil
- *
- *  Announce an intention to access data from the current file in a
- *  specific pattern. On platforms that do not support the
- *  <em>posix_fadvise(2)</em> system call, this method is a no-op.
+ *    advise(advice, offset = 0, len = 0) -> nil
  *
- *  _advice_ is one of the following symbols:
+ *  Announces an intention to access data from the current file in a specific pattern.
  *
- *  :normal::     No advice to give; the default assumption for an open file.
- *  :sequential:: The data will be accessed sequentially
- *                with lower offsets read before higher ones.
- *  :random::     The data will be accessed in random order.
- *  :willneed::   The data will be accessed in the near future.
- *  :dontneed::   The data will not be accessed in the near future.
- *  :noreuse::    The data will only be accessed once.
+ *  Has no effect  on a platform that does not support system call
+ *  {posix_fadvise(2)}[https://linux.die.net/man/2/posix_fadvise].
  *
- *  The semantics of a piece of advice are platform-dependent. See
- *  <em>man 2 posix_fadvise</em> for details.
+ *  The relevant data is specified by:
  *
- *  "data" means the region of the current file that begins at
- *  _offset_ and extends for _len_ bytes. If _len_ is 0, the region
- *  ends at the last byte of the file. By default, both _offset_ and
- *  _len_ are 0, meaning that the advice applies to the entire file.
+ *  - +offset+: The offset of the first byte of data.
+ *  - +len+: The number of bytes to be accessed;
+ *    if +len+ is zero, or is larger than the number of bytes remaining,
+ *    all remaining bytes will be accessed.
  *
- *  If an error occurs, one of the following exceptions will be raised:
+ *  Argument +advice+ is one of the following symbols:
```

Comment:
TBH, this formulation from the original docs seems pretty valuable to me:
> The semantics of a piece of advice are platform-dependent. 

Can we maybe restore it?..

Reply:
This seems not to be addressed

Reply:
Modified.

Reply:
Modified.